### ISSUE TITLE
SVGs render to an empty PNG

### ISSUE THREAD

**Issue by texzone** [Kozea/CairoSVG#399]
I have a couple of SVG files that I wish to convert to PNG or JPEG - doesn't matter really. Unfortunately, whenever I try to render them to JPG or PNG using CairoSVG, the pictures are empty (contain nothing). I am not sure where the issue is, but actually, when I try to do it online using any other SVG -> PNG tools, the same thing happens. Would love some guidance as to why this is happening.

I have included three SVG files here as an example that you can see in SVG-Viewer.

[N-Sqaured](https://www.svgviewer.dev/s/i90w8yIl)
[Additions](https://www.svgviewer.dev/s/WG1fubcd)
[3x3](https://www.svgviewer.dev/s/nOUOc6TW)

So, yeah - what could be the reason for this issue?

**Comment by liZe** [comment]
Hi!

Your files work well for me:
![ex](https://github.com/Kozea/CairoSVG/assets/952243/a4c362f1-650c-4211-8ffc-ebb01a5b91d1)
![ex2](https://github.com/Kozea/CairoSVG/assets/952243/31216767-4d8b-466b-bc77-9ef3f48b23e9)

I’ve tried CairoSVG 2.6.0 and 2.7.1, they both work. Do you use an older version?

**Comment by texzone** [comment]
> Hi!
> 
> Your files work well for me: ![ex](https://user-images.githubusercontent.com/952243/262630630-a4c362f1-650c-4211-8ffc-ebb01a5b91d1.png) ![ex2](https://user-images.githubusercontent.com/952243/262630660-31216767-4d8b-466b-bc77-9ef3f48b23e9.png)
> 
> I’ve tried CairoSVG 2.6.0 and 2.7.1, they both work. Do you use an older version?

Huh, interesting! I am using CairoSVG version 2.7.1 on Windows. I did experience a lot of trouble getting it installed though, and ultimately had to install the libcairo-2.dll from some other source and use that DLL. Could that be the issue, that this particular DLL is outdated?

**Comment by texzone** [comment]
Interestingly enough, I have deleted uiconverter, gtk3, and deleted reference of them from my systempath, and uninstalled cairosvg then restarted my PC. When I reinstall, I dont get the same error, but still get the same output (empty PNG)

**Comment by liZe** [comment]
> I feel like all I need it the .dll file.

Yes, you need the DLL files, and the GTK3 installer is probably the easiest way to get them.

> and still.... nothing.

Do you mean that the error is gone but that the image is empty?

**Comment by texzone** [comment]
Yes, the image is still empty basically empty.

```
> python
Python 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import cairosvg
>>> cairosvg.__version__
'2.7.1'
```

This is the code im running:

```
def svg_to_png_jpeg(input_svg_path, output_png_path):
    cairosvg.svg2png(
        url=input_svg_path,
        write_to=output_png_path,
        background_color="white",
        scale=2
    )
    ```
    
    These are the images I get:
    [With background=white](https://cdn.discordapp.com/attachments/992171308563566783/1144004187638739035/test.png)
    [With no background](https://cdn.discordapp.com/attachments/992171308563566783/1144004118982168707/test.png)
    
    ... issues with AMD?

**Comment by liZe** [comment]
> ... issues with AMD?

CairoSVG does not use the graphic card.

Do you have the same problem by launching `python -m cairosvg --background=white --scale=2 input.svg -o output.png`?

**Comment by texzone** [comment]
Yes, unfortunately, I do...

**Comment by liZe** [comment]
> So you dont think the issue is with a... weird `.dll` file im getting for GTK3?

I don’t think so. I really don’t know where it could come from.

Could you please share the result of `python -m cairosvg --background=white --scale=2 input.svg -o output.svg`?

**Comment by texzone** [comment]
Yes, absolutely. Here are the files

![output.png](https://github.com/Kozea/CairoSVG/assets/12116349/6c0dbc4d-e68a-4872-aa90-b0360f61ea17)
![input.svg](https://github.com/Kozea/CairoSVG/assets/12116349/d7ebd7d8-1447-45f1-8582-38dc4e0cc497)

**Comment by texzone** [comment]
Shoot, my bad, I just reread your comment. The output needs to be SVG, not PNG. My bad.
![output.svg](https://github.com/Kozea/CairoSVG/assets/12116349/d03228bf-f5ed-47e7-b575-43a747b47287)

There is the svg file. My bad

**Comment by liZe** [comment]
I really don’t know where your problem comes from. I’ve tried two different versions of Cairo on two different computers, and it works well for me.

> I am not sure where the issue is, but actually, when I try to do it online using any other SVG -> PNG tools, the same thing happens.

That’s really, really strange, because your file is pretty easy SVG and should give correct results with many converters. Are you sure that the input SVG are exactly the ones you provided in this issue? Is the original SVG correctly displayed in a browser on your computer for example?

The only "strange" thing in this files is the "ex" unit used in the `svg` tag for width and height. Do you get the same result when you use `px` instead of `ex`?

**Comment by texzone** [comment]
I truly have no idea... well, I'm going to go bang my head against the wall for a bit and see if any divine intervention occurs. Thank you so much for your time man!
